src/network/networkd-network.c

```diff
@@ -1321,11 +1321,17 @@ int config_parse_required_for_online(
                 }
 
                 required = r;
-                s = LINK_OPERSTATE_DEGRADED;
+                range = LINK_OPERSTATE_RANGE_DEFAULT;
         }
 
         network->required_for_online = required;
-        network->required_operstate_for_online = s;
+
+        network->required_operstate_for_online.min = range.min != _LINK_OPERSTATE_INVALID
+                                                                ? range.min
+                                                                : LINK_OPERSTATE_RANGE_DEFAULT.min;
+        network->required_operstate_for_online.max = range.max != _LINK_OPERSTATE_INVALID
+                                                                ? range.max
+                                                                : LINK_OPERSTATE_RANGE_DEFAULT.max;
```

Comment:
This is still wrong. ":dormant" should mean anything between "missing" (the lowest state) and "dormant", but right now this is the same as "degraded:dormant" (which doesn't even make sense). I.e. please just do the same change as in the other place.

Reply:
I moved the default value logic into `parse_operational_state_range`. The only thing it doesn't handle is empty operational state ranges as those have to be handled differently depending on whether we're parsing unit code or operational state ranges passed on the command line. In unit code, we default to degraded:routable if an operational state range is empty. In wait-online, we error on empty operational state ranges. (`-i eth1` is fine, `-i eth1:` is not)